### ISSUE TITLE
Unable to export dataframe/list to Excel with xlwings

### ISSUE THREAD

**Issue by MarekSevela** [xlwings/xlwings#1606]
#### OS (e.g. Windows 10 or macOS Sierra)
Windows 10

#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)
v0.23.3

#### Describe your issue (incl. Traceback!)
Hello Felix,

I have a daily scheduled script exporting Outlook emails to Excel, for the actual exporting to Excel, I am using xlwings because I want to keep formatting and other things intact in the file. (wasnt lucky setting it up with xlsxwriter that way)

Script was running for a few weeks without any exporting issues but recently I cannot see full data exported in the Excel after completion. Only a few random lines throughout Excel rows appear and there is no error message in the code.

When exporting Outlook data, I am first gathering them into a python list, then I export that list with xlwings in this fashion:

`TargetReport_Workbook.sheets['Last Daily Export'].range('A2').options(header=False,index=False,strings_to_urls= False).value = list_to_dataframe`

I was searching for solution and the only one I found is this:
https://stackoverflow.com/questions/56593684/exporting-dataframe-with-xlwings-without-having-a-index-column

Basically saying that there could still be some issues with python lists and how xlwings handles them.

For that I tried converting my list to pandas dataframe and export it afterwards, as suggested, but I am receiving the same results.

I am quite certain that there is no issue in my code, because there was no change to it recently + when I try to export it with xlsxwriter, to check the data, its looking completely fine.

May I please ask for you guidance on this?

Thank you very much and keep up the great work! Xlwings is my daily tool of choice for quite some time now. :)

Best Regards,

Mark

**Comment by ericbmoreira** [comment]
I cannot think of why it would break if nothing has changed. I'm a little confused about the "list" though. What it looks like you are pasting is a dataframe. Any possibility of pasting the data that you are trying to use in this thread? I know that pandas can infer some data types and that can mess things up when you try to paste it into Excel. Also the formatting in Excel can cause problems.

Have you tried this to see what happens? TargetReport_Workbook.sheets['Last Daily Export'].range('A2').value = list_to_dataframe

It certainly sounds like something in the data is causing some issues when you bring it into Python. How do you like to debug? I like to use a terminal session, not a debug session, and manually go through each line and see what it prints. Then a nice thing to use is print(df.to_string()) and it will print out the entire df in your terminal so you can look at it.

**Comment by MarekSevela** [comment]
Hello ericbmoreira,

thank you for your time. The reason I think there might be something with the xlwings exporting is that when replacing xlwings exporting with xlsxwriter, that creates new file and just puts it in, there is nothing wrong with the data. If there would be some formatting or other issues along the code, it would be visible in xlsxwriter export. (I believe)

Unfortunately, I cannot provide you the data as these are sensitive company email conversations I am working with.

I tried your suggestion - exporting just with `TargetReport_Workbook.sheets['Last Daily Export'].range('A2').value = list_to_dataframe` but I am getting the same results, missing columns, some of them half empty etc.

Is there a way to use xlsxwriter to just paste the data into an existing Excel Workbook without creating a new one? Would not solve my issue with xlwings export but I could at least move on with the project.

Thank you very much for helping!

**Comment by ericbmoreira** [comment]
It just seems strange that is used to work. Since you can't show the data how about the code snippet from the point it captures the email data and coverts it to the dataframe and pastes it? Maybe also the df.info() data? That will at least show what the types are. Can you try to paste it into a different sheet just for testing? I've never run into a missing data problem like this before. It seems weird that the dataframe is fine up to the point of sending it to Excel.

**Comment by fzumstein** [comment]
Just two observations: when you gather your data in a list, there is no need to transform it into a DataFrame first (xlwings will transform it back to a list anyways). Also, what about that `strings_to_urls` argument? Not that this would explain the issue but it's not a supported option. If you want us to be able to help, you'd need to put together a dummy sample that reproduces the issue, otherwise it's difficult. Double-check if your template changed and you have any active filters or something, but the StackOverflow issue has nothing to do with your issue.
